packages/sdk/src/rpcInterceptors.ts

```diff
@@ -78,6 +78,20 @@ export const retryInterceptor: (retryParams: RetryParams) => Interceptor = (
         }
 }
 
+export const expiryInterceptor = (opts: { onTokenExpired?: () => void }): Interceptor => {
+    return (next) => async (req) => {
+        try {
+            const res = await next(req)
+            return res
+        } catch (e) {
+            if (errorContains(e, Err.PERMISSION_DENIED) && req.method.name === 'AddEvent') {
+                opts.onTokenExpired?.()
+            }
```

Comment:
I really like the interceptor. This condition isn't correct though. There are more places than just AddEvent that throw... can you check for the string "event delegate has expired"? if not then we need a new error on the nodes.

Reply:
Nice, event delegate has expired string matches what we want here. Would be great to have the error on the node itself, but I would like to model this first https://github.com/river-build/river/pull/1222#discussion_r1801194160 and then clean the remaining things. wdyt?